tests/Fixtures/CodeSamples/ReturnTypes.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Infection\CodeSamples;
+
+class ReturnTypes
+{
+    public function withReturnType(): int
+    {
+        return 3;
+    }
+
+    public function nullableReturnType(): ?int
+    {
+        return null;
+    }
+
+    public function withoutReturnType()
+    {
+        return 3;
+    }
+
+    public function selfReturn(): self
```

Comment:
https://3v4l.org/2tQLa

it's possible to have `parent` as a return typehint, as well as a parameters typehint

Reply:
Cool, didn't know that was an option in PHP, ill add it gere